### PR TITLE
"How to make a threaded chat" tutorial

### PR DESCRIPTION
**Pull request by keerlu**

### CODE REVIEW

File: howtos/how-to-make-a-threaded-chat/index.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          "replyto_" + event.message.id
+        );
+        thread.setParticipant(me);
+        chatbox.select(thread);
+      });
+
+      chatbox.onCustomConversationAction("back", async (event) => {
+        const parentConvId = event.conversation.custom.parentConvId;
+
+        if (parentConvId != undefined) {
```
Comment (MattivdWeem) [inline howtos/how-to-make-a-threaded-chat/index.html:93]: We can probably write this as `if(!parentConvId){}` to remove the `undefined` or write it as `parentConvId !== undefined`

File: howtos/how-to-make-a-threaded-chat/server.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+app.use(express.json());
+
+app.listen(3000, () => console.log("Server is up"));
+
+const senderId = `threadsExampleSender`;
+const receiverId = `threadsExampleReceiver`;
+
+// Create a thread as a new conversation
+async function createThread(parentMessageId, parentConvId, participants) {
+  const conversationId = "replyto_" + parentMessageId;
```
Comment (MattivdWeem) [inline howtos/how-to-make-a-threaded-chat/server.js:21]: It's a bit confusing that we use both concatenation and interpolation in the same code same for using both backticks and double quotes

File: howtos/how-to-make-a-threaded-chat/server.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${secretKey}`,
+      },
+    }
+  );
+}
+
+app.post("/newThread", async (req, res) => {
+  // Get details of the message we'll reply to
+  const parentMessageId = req.body["messageId"];
```
Comment (MattivdWeem) [inline howtos/how-to-make-a-threaded-chat/server.js:92]: I'd use `.messageId` here like we do in other parts of the code

File: howtos/how-to-make-a-threaded-chat/server.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+  const parentConvId = req.body["conversationId"];
+  const parentMessageText = req.body["messageText"];
+  const parentParticipants = req.body["participants"];
+
+  let response = await getMessages(parentMessageId);
+  let messages = await response.json();
+
+  // Create a message with the text of the parent message if one doesn't already exist
+  if (messages.data === undefined || messages.data.length == 0) {
```
Comment (MattivdWeem) [inline howtos/how-to-make-a-threaded-chat/server.js:101]: you could check this with `!messages.data || !messages.data.length` or change the == into a ===

File: howtos/how-to-make-a-threaded-chat/server.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+}
+
+app.post("/newThread", async (req, res) => {
+  // Get details of the message we'll reply to
+  const parentMessageId = req.body["messageId"];
+  const parentConvId = req.body["conversationId"];
+  const parentMessageText = req.body["messageText"];
+  const parentParticipants = req.body["participants"];
+
+  let response = await getMessages(parentMessageId);
```
Comment (MattivdWeem) [inline howtos/how-to-make-a-threaded-chat/server.js:97]: No need for both the message and the response to be a let

File: howtos/how-to-make-a-threaded-chat/server.js [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+});
+
+// Endpoint for message.sent webhook
+app.post("/updateReplyCount", async (req, res) => {
+  const data = req.body.data;
+  const conversationId = data.conversation.id;
+  const messageType = data.message.type;
+
+  if (
+    conversationId.slice(0, 8) === "replyto_" &&
```
Comment (MattivdWeem) [inline howtos/how-to-make-a-threaded-chat/server.js:116]: a neat trick might be to use `"replyto_".startsWith("replyto_")` for readability

File: howtos/how-to-make-a-threaded-chat/server.js [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+    messageType === "UserMessage"
+  ) {
+    const parentMessageId = data.conversation.custom.parentMessageId;
```
Comment (MattivdWeem) [inline howtos/how-to-make-a-threaded-chat/server.js:119]: you might be able to decosntruct this 

```
const {parentMessageId, parentConvId} = data.conversaton.custom;
```

**Review by MattivdWeem — COMMENTED** [review]
Minor nitpicking about some code, loving it otherwise!

File: howtos/how-to-make-a-threaded-chat/index.html (hunk 1 above)
Comment (keerlu) [reply howtos/how-to-make-a-threaded-chat/index.html]: Updated with `!== undefined`

File: howtos/how-to-make-a-threaded-chat/server.js (hunk 3 above)
Comment (keerlu) [reply howtos/how-to-make-a-threaded-chat/server.js]: Done

File: howtos/how-to-make-a-threaded-chat/server.js (hunk 4 above)
Comment (keerlu) [reply howtos/how-to-make-a-threaded-chat/server.js]: Updated to `!messages.data?.length`

File: howtos/how-to-make-a-threaded-chat/server.js (hunk 5 above)
Comment (keerlu) [reply howtos/how-to-make-a-threaded-chat/server.js]: Updated to `const`

File: howtos/how-to-make-a-threaded-chat/server.js (hunk 6 above)
Comment (keerlu) [reply howtos/how-to-make-a-threaded-chat/server.js]: Definitely nicer!

File: howtos/how-to-make-a-threaded-chat/server.js (hunk 7 above)
Comment (keerlu) [reply howtos/how-to-make-a-threaded-chat/server.js]: Done!

File: howtos/how-to-make-a-threaded-chat/README.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+This is an example project for TalkJS's tutorial on how to make a comment section with threaded replies.
+
+!! expand...
```
Comment (keerlu) [inline howtos/how-to-make-a-threaded-chat/README.md:3]: Finish summary

File: howtos/how-to-make-a-threaded-chat/server.js (hunk 2 above)
Comment (keerlu) [reply howtos/how-to-make-a-threaded-chat/server.js]: Switched to all interpolation, that's definitely better

**Pull request merged**